llarp/service/endpoint.cpp

```diff
@@ -86,18 +86,19 @@ namespace llarp
           ManualRebuild(1);
         return;
       }
-      introSet().I.clear();
+      std::vector< Introduction > intros;
       for(auto& intro : I)
       {
-        introSet().I.emplace_back(std::move(intro));
+        intros.emplace_back(std::move(intro));
```

Comment:
I doubt the `std::move()` is doing anything useful here (yes, I realize that's not new to this change). Changing `GetCurrentIntrductionsWithFilter()` to take a `std::set` or changing `introSet().I` to be a `std::vector` might avoid extra copies/logic.

Then again, this isn't a hot path and `working - optimized` > `optimized - working`